src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/BlockPlacer.java

```diff
@@ -182,6 +186,10 @@ private void placeSlimefunBlock(SlimefunItem sfItem, ItemStack item, Block block
     private void placeBlock(ItemStack item, Block facedBlock, Dispenser dispenser) {
         BlockPlacerPlaceEvent e = new BlockPlacerPlaceEvent(dispenser.getBlock(), item, facedBlock);
         Bukkit.getPluginManager().callEvent(e);
+        
+        if(!dispenser.getInventory().getViewers().isEmpty()){
+        	e.setCancelled(true);
```

Comment:
There is no point in calling an Event when it gets cancelled anyway.
The whole point of the Event contract is that **others** get a say in the outcome of the Event, not that their changes get overridden otherwise.

If no block gets placed, then there shouldn't even be an Event called.

Reply:
Sorry, but the event has been called at Bukkit.getPluginManager().callEvent(e).

Reply:
But if add if before it ,maybe it will be slower?
Because most of the time, dispenser.getInventory().getViewers().isEmpty() is true.